README.adoc

```diff
@@ -78,29 +78,79 @@ checkout([$class: 'GitSCM',
           userRemoteConfigs: [[url: 'https://github.com/jenkinsci/git-client-plugin.git']]])
 ---
 
-==== Checkout with ssh and a private key credential
+==== Checkout with a shallow clone to reduce data traffic
 
-Insert example here
+Checkout from the git client plugin source repository using https no credentials, a default branch, and with a shallow clone which  requests a limited number of commits from the tip of the requested branch(es) thereby saving time, data transfer and disk space.
 
-==== Checkout with https and a username/password credential
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: '*/master']],
+          extensions: [[$class: 'CloneOption', noTags: false, reference: '', shallow: true]],
+          userRemoteConfigs: [[url: 'https://github.com/jenkinsci/ws-cleanup-plugin']]])
+---
 
-Insert example here
+==== Checkout with a narrow refspec
 
-==== Checkout with git large file support enabled
+Checkout from the git client plugin source repository using https no credentials, a default branch, and with refspec defined for the repository. This can save time, data transfer and disk space when you only need to access the references specified by the refspec.
+
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: '*/master']],
+          extensions: [[$class: 'CloneOption', noTags: false, reference: '', shallow: false, honorRefspec: true]],
+          userRemoteConfigs: [[url: 'https://github.com/jenkinsci/ws-cleanup-plugin']]])
+---
 
-Enable git lfs pull after checkout, create the snippet
+==== Checkout and prune stale remote branches
 
-==== Add a repository browser for Bitbucket
+Checkout from the git client plugin source repository using https no credentials, a default branch, and with prune tags extension enabled. This removes remote tracking branches from the local workspace if they no longer exist on the remote.
 
-==== Checkout without fetching tags (advanced clone behavior)
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: '*/master']],
+          extensions: [[pruneTags(true)]],
+          userRemoteConfigs: [[url: 'https://github.com/jenkinsci/ws-cleanup-plugin']]])
+---
 
-==== Checkout with a shallow clone to reduce data traffic
+==== Clean after checkout and delete untracked nested repositories
 
-==== Checkout with a narrow refspec (needs more reading)
+Checkout from the git client plugin source repository using https no credentials, a default branch, and clean checkout extension with delete untracked nested repositories enabled. Clean the workspace after every checkout by deleting all untracked files and directories, including those which are specified in .gitignore.
```

Comment:
Make the sentence match the example.  This repository prefers sentence per line formatting in the documentation, so inserted a line break at the end of the sentence.
```suggestion
Checkout from the workspace cleanup plugin source repository using https no credentials, a default branch, and clean checkout extension with delete untracked nested repositories enabled.
Clean the workspace after every checkout by deleting all untracked files and directories, including those which are specified in the `.gitignore` file.
```